src/AppEntry.tsx

```diff
@@ -65,7 +67,8 @@ const AppWithKeycloak: React.FunctionComponent = () => {
 
 const AppWithConfig: React.FunctionComponent = () => {
   const config = useContext(ConfigContext);
-  if (config === undefined) {
+  const constants =  useContext(ConstantContext);
+  if (config === undefined || constants === undefined) {
```

Comment:
Pausing the app loading until a JSON file is loaded that isn't needed to much deeper in the app execution is going to uncessarily impact loading performance. Instead you need to include an async get method on the context and call that as needed in a page/view.

Config is different as loading can't progress without. Note this is one of those "thousand papercuts" - on it's own not a major problem, but add them all up and they cause a real problem.

Reply:
Yes. That was quite bad part on me. I have done refactor as you have mentioned but found that excessively overkill for use cases so simplified further. Hope that works.